kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -764,6 +810,140 @@ namespace Kratos
+                        all_neighbour_set.insert(neigh);
+                    }
+
+                    bool is_edge_cut = false;
+                    for (auto& g_ptr_neigh : all_neighbour_set) {
+                        if (!is_edge_cut) {
+                            auto cut_edges = get_neighbour_cut_edges.Get(g_ptr_neigh);
+                            for (auto cut_edge: cut_edges) {
+                                if (cut_edge[0]==edge_i_id && cut_edge[1]==edge_j_id) {
+                                    is_edge_cut = true;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+
+                    // Correcting edge if not intersected after applying an epsilon
+                    if (!is_edge_cut) {
+                        rCuteEdgeVector[i_edge] = -1;
+                    }
+                }
+            }
+        };
+
+        for (auto &r_elem : mrVolumePart.Elements())  {
+            const auto edges_container = r_elem.GetGeometry().GenerateEdges();
+            auto &r_cut_edge_vector = r_elem.GetValue(ELEMENTAL_EDGE_DISTANCES);
+            check_edge_against_distances(edges_container, r_cut_edge_vector);
+            if (mOptions.Is(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EDGE_DISTANCES_EXTRAPOLATED))
+            {
+                auto &r_cut_edge_extra_vector = r_elem.GetValue(ELEMENTAL_EDGE_DISTANCES_EXTRAPOLATED);
+                check_edge_against_distances(edges_container, r_cut_edge_extra_vector);
+            }
+        }
+
+        KRATOS_CATCH(" ");
+    }
+
+    template <std::size_t TDim>
+    void CalculateDiscontinuousDistanceToSkinProcess<TDim>::CalculateElementalNeighbours()
+    {
+        KRATOS_TRY;
+
+        FindGlobalNodalElementalNeighboursProcess find_nodal_elems_process(mrVolumePart);
+        find_nodal_elems_process.Execute();
+
+        KRATOS_CATCH(" ");
+    }
+
+    template<std::size_t TDim>
+    GlobalPointerCommunicator<Element>::Pointer CalculateDiscontinuousDistanceToSkinProcess<TDim>::CreatePointerCommunicator()
+    {
+        KRATOS_TRY;
+
+
+        auto &r_comm = mrVolumePart.GetCommunicator().GetDataCommunicator();
+
+        std::vector<int> elem_indices;
+        elem_indices.reserve(mrVolumePart.NumberOfElements());
+        for (auto& r_elem : mrVolumePart.Elements()) {
+            elem_indices.push_back(r_elem.Id());
+        }
+
+        auto g_ptr_elem_map = GlobalPointerUtilities::RetrieveGlobalIndexedPointersMap(mrVolumePart.Elements(), elem_indices, r_comm);
+
+        GlobalPointersVector<Element> g_ptr_elem_list;
+        for(auto& item: g_ptr_elem_map) {
+            g_ptr_elem_list.push_back(item.second);
+        }
+
+        for (auto& r_node : mrVolumePart.Nodes()) {
+            for (auto& g_ptr : r_node.GetValue(NEIGHBOUR_ELEMENTS).GetContainer()) {
```

Comment:
If I'm not wrong, this is a reference to a pointer. I'd avoid that.

Reply:
CLang seems to not like this:

```
In file included from /__w/Kratos/Kratos/build/Custom/kratos/CMakeFiles/KratosCore.dir/Unity/unity_2_cxx.cxx:5:
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:865:36: error: loop variable 'p_neigh' of type 'const Kratos::GlobalPointer<Kratos::Element>' creates a copy from type 'const Kratos::GlobalPointer<Kratos::Element>' [-Werror,-Wrange-loop-construct]
                    for(const auto p_neigh : r_edge_containter_0) {
                                   ^
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:865:25: note: use reference type 'const Kratos::GlobalPointer<Kratos::Element> &' to prevent copying
                    for(const auto p_neigh : r_edge_containter_0) {
                        ^~~~~~~~~~~~~~~~~~~~
                                   &
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:869:36: error: loop variable 'p_neigh' of type 'const Kratos::GlobalPointer<Kratos::Element>' creates a copy from type 'const Kratos::GlobalPointer<Kratos::Element>' [-Werror,-Wrange-loop-construct]
                    for(const auto p_neigh : r_edge_containter_1) {
                                   ^
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:869:25: note: use reference type 'const Kratos::GlobalPointer<Kratos::Element> &' to prevent copying
                    for(const auto p_neigh : r_edge_containter_1) {
                        ^~~~~~~~~~~~~~~~~~~~
                                   &
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:874:37: error: loop variable 'g_ptr_neigh' of type 'const Kratos::GlobalPointer<Kratos::Element>' creates a copy from type 'const Kratos::GlobalPointer<Kratos::Element>' [-Werror,-Wrange-loop-construct]
                    for (const auto g_ptr_neigh : all_neighbour_set) {
                                    ^
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:874:26: note: use reference type 'const Kratos::GlobalPointer<Kratos::Element> &' to prevent copying
                    for (const auto g_ptr_neigh : all_neighbour_set) {
                         ^~~~~~~~~~~~~~~~~~~~~~~~
                                    &
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:930:29: error: loop variable 'g_ptr' of type 'const Kratos::GlobalPointer<Kratos::Element>' creates a copy from type 'const Kratos::GlobalPointer<Kratos::Element>' [-Werror,-Wrange-loop-construct]
            for (const auto g_ptr : r_node.GetValue(NEIGHBOUR_ELEMENTS).GetContainer()) {
                            ^
/__w/Kratos/Kratos/kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:930:18: note: use reference type 'const Kratos::GlobalPointer<Kratos::Element> &' to prevent copying
            for (const auto g_ptr : r_node.GetValue(NEIGHBOUR_ELEMENTS).GetContainer()) {
                 ^~~~~~~~~~~~~~~~~~
                            &
```